shortfin/tests/apps/sd/serving_test.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import requests
+import time
+import pytest
+import subprocess
+import os
+import socket
+import sys
+import copy
+from contextlib import closing
+import numpy
+from PIL import Image
+
+from shortfin_apps.types.Base64CharacterEncodedByteSequence import (
+    Base64CharacterEncodedByteSequence,
+)
+
+from shortfin_apps.utilities.image import (
+    image_from,
+)
+
+BATCH_SIZES = [1]
+
+sample_request = {
+    "prompt": [
+        " a cat under the snow with blue eyes, covered by snow, cinematic style, medium shot, professional photo, animal",
+    ],
+    "neg_prompt": ["Watermark, blurry, oversaturated, low resolution, pollution"],
+    "height": [1024],
+    "width": [1024],
+    "steps": [20],
+    "guidance_scale": [7.5],
+    "seed": [0],
+    "output_type": ["base64"],
+    "rid": ["string"],
+}
+
+
+def start_server(fibers_per_device=1, isolation="per_fiber"):
+    # Start the server
+    srv_args = [
+        "python",
+        "-m",
+        "shortfin_apps.sd.server",
+    ]
+    with open("sdxl_config_i8.json", "wb") as f:
+        r = requests.get(
+            "https://sharkpublic.blob.core.windows.net/sharkpublic/sdxl/04082025/configs/sdxl_config_i8.json",
+            allow_redirects=True,
+        )
+        f.write(r.content)
+    srv_args.extend(
+        [
+            f"--model_config=sdxl_config_i8.json",
+            f"--fibers_per_device={fibers_per_device}",
+            f"--isolation={isolation}"
+        ]
+    )
+    runner = ServerRunner(srv_args)
+    # Wait for server to start
+    time.sleep(3)
+    return runner
+
+
+@pytest.fixture(scope="module")
+def sd_server_fpd1():
+    runner = start_server(fibers_per_device=1)
+
+    yield runner
+
+    if os.path.exists("sdxl_config_i8.json"):
+        os.remove("sdxl_config_i8.json")
```

Comment:
Oh, this is writing into the source directory? If keeping the file download, prefer to use a temp directory (pytest has utilities for this), so there is no risk of leaving a loose file in the source directory that could accidentally get committed to git.

(also prefer to use pathlib over `os.path`)